api/fossa/upload.go

```diff
@@ -32,15 +32,15 @@ type UploadOptions struct {
 
 // Upload uploads a project's analysis.
 func Upload(title string, locator Locator, options UploadOptions, data []SourceUnit) (Locator, error) {
-	// Check preconditions
+	// Check preconditions.
 	if locator.Fetcher != "custom" && locator.Revision == "" {
```

Comment:
This makes a little more sense as `if locator.Fetcher == "git" && locator.Revision == "" {`.